common/src/main/scala/org/apache/celeborn/common/serializer/JavaSerializer.scala

```diff
@@ -98,19 +99,49 @@ private[celeborn] class JavaSerializerInstance(
 
   override def serialize[T: ClassTag](t: T): ByteBuffer = {
     val bos = new ByteBufferOutputStream()
+    val msg = Utils.toTransportMessage(t)
+    msg match {
+      case transMsg: TransportMessage =>
+        // Check if the msg is a TransportMessage with language-agnostic V2 serdeVersion.
+        // If so, write the marker and the body explicitly.
+        if (transMsg.getSerdeVersion == SerdeVersion.V2) {
```

Comment:
I understand this is the minimal change, though it's a little weird to write that logic inside `JavaSerializer*`. If you don't want to touch much Java/Scala code, I'm OK with the current state.